crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -1461,10 +1462,11 @@ impl CheckpointSignatureAggregator {
             "Checking for split brain condition"
         );
         if self.signatures_by_digest.quorum_unreachable() {
-            // TODO: at this point we should immediately halt processing
-            // of new transaction certificates to avoid building on top of
-            // forked output
-            // self.halt_all_execution();
+            // At this point we should immediately halt processing
+            // of new transaction certificates to avoid building on
+            // top of forked output
+            let state_clone = self.state.clone();
+            spawn_monitored_task!(halt_all_execution(state_clone));
```

Comment:
Curious why this needs to be async?

Reply:
I have a RW lock to control access to the in-memory execution state within authority state, as it could be possible that multiple tasks have a reference to authority state